cmd/kubeadm/app/discovery/token/token.go

```diff
@@ -171,6 +123,74 @@ func RetrieveValidatedConfigInfo(cfg *kubeadmapi.JoinConfiguration) (*clientcmda
 	return baseKubeConfig, nil
 }
 
+//getInsecureKubeconfigBytes is used to return insecureKubeconfigBytes from insecureBootstrapConfig and token
+func getInsecureKubeconfigBytes(insecureBootstrapConfig *clientcmdapi.Config, token *kubeadmapi.BootstrapTokenString) ([]byte, error) {
+	insecureClient, err := kubeconfigutil.ToClientSet(insecureBootstrapConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// Make an initial insecure connection to get the cluster-info ConfigMap
+	var insecureClusterInfo *v1.ConfigMap
+	wait.PollImmediateInfinite(constants.DiscoveryRetryInterval, func() (bool, error) {
+		var err error
+		insecureClusterInfo, err = insecureClient.CoreV1().ConfigMaps(metav1.NamespacePublic).Get(bootstrapapi.ConfigMapClusterInfo, metav1.GetOptions{})
+		if err != nil {
+			klog.V(1).Infof("[discovery] Failed to request cluster info, will try again: [%s]\n", err)
+			return false, nil
+		}
+		return true, nil
+	})
+
+	// Validate the MAC on the kubeconfig from the ConfigMap and load it
+	insecureKubeconfigString, ok := insecureClusterInfo.Data[bootstrapapi.KubeConfigKey]
```

Comment:
From the start of the func (the call of `ToClientSet`) to this line (inclusive), the code looks mostly the same with the start of `getSecureKubeconfig`.
Can you extract the common part in another helper (to be used by these couple of funcs)?

Reply:
Thanks for your reminding, I have modified.